core/src/main/scala/kafka/coordinator/group/GroupMetadataManager.scala

```diff
@@ -614,9 +615,13 @@ class GroupMetadataManager(brokerId: Int,
               }
             }
             currOffset = batch.nextOffset
+            batch.iterator().forEachRemaining(record => {
+              bytesRead += record.sizeInBytes()
+              numMessagesRead += 1
```

Comment:
You should probably do this in the existing loop where we iterate over the records. We don't want to decompress the batch a second time.

Reply:
@ijuma I did consider that. Infact, even I didn't want to decompress the records a second time. The only issue there is it might lead to code duplicity because the way the records are iterated within the if and else blocks.

Reply:
@ijuma I moved the iteration loop inside existing iterations, as you suggested.